src/repository/impl/repositoryimpl.ts

```diff
@@ -0,0 +1,199 @@
+import { AnyBulkWriteOperation, Model, Types } from "mongoose";
+import {
+  manyUpdateResponse,
+  UpdateResponse,
+} from "../../Dto/Response/updateRes";
+import { PhoneEntry } from "../../Dto/Request/createReq";
+import { GetResponse } from "../../Dto/Response/getRes";
+import {
+  UpdateManyPhoneEntry,
+  UpdatePhoneEntry,
+} from "../../Dto/Request/updateReq";
+import { DataDocument, Repo } from "../repository";
+import { manyEntry } from "../../Dto/Request/createReq";
+import { manyResponse } from "../../Dto/Response/createRes";
+import { paginationresponse } from "../../Dto/Response/getRes";
+import { Response } from "../../Dto/Response/createRes";
+import { error } from "winston";
+
+class SchemaRepository implements Repo {
+  async getData(dir: Model<DataDocument>): Promise<GetResponse[] | undefined> {
+    const data = await dir.find().exec();
+    throw new Error("asfaga");
+    return data;
+  }
+  async addData(body: PhoneEntry, dir: Model<DataDocument>): Promise<Response> {
+    let now = new Date();
+    const newData = new dir({
```

Comment:
create a mapper class which will have method to use this in service layer and send only mongo object to repository
1. map dto -> mongoObject
2. mongoObject -> dto

Reply:
done

Reply:
But I have reused the two functions inside the map function for all the requirements. Is that okay?

Reply:
i am still working on the logging the information inside the terminal

Reply:
what should be the casing for the files inside the package